### ISSUE TITLE
Choro Q HG 2 (Road Trip Adventure) music corrupted when exported (PS2 Tamsoft format)

### ISSUE THREAD

**Issue by TheShinyCookie** [vgmtrans/vgmtrans#138]
## Expected Behavior
I'm wanting to convert the music for Road Trip Adventure into MIDIs and soundfonts, which I can play around with. Ideally, it should be pretty close to the original (of course I don't expect it to be perfect). [Here's what Peach Raceway should sound like.](https://youtu.be/Tq2RdwC100Y)

## Current Behavior
Playing the track in vgmtrans results in [this](https://ufile.io/rdki5). The music is off key and the reverb isn't emulated properly, but it's still listenable. (Either the game handled reverb differently, or it's something to do with ADSR or something like that; there's still a lot of reverb, but it doesn't sound right.)

Exporting it to MIDI and playing it in a MIDI player results in [this mess](https://ufile.io/9uh5k). This, I cannot listen to. This happens in any MIDI player, with both soundfonts and DLS files.

## Possible Solution
I'm not a developer, so I have absolutely no idea where the problem lays.

## Steps to Reproduce (for bugs)
1. Rip the Choro Q HG 2 disc
2. Extract the music folder
3. Rename BGM.TVB to BGM.TVB2.
4. Open a BGM_**.TSQ file, and then BGM.TVB2 (don't know why it has to be done in this order...)
5. Convert the music from the Collections window.
6. Open both of the resulting files in a MIDI player and enjoy the cacophony (or not).

## Context
Of course, I can't convert the music to MIDI because it doesn't work right. I was hoping to play around with it in a MIDI editor as well.

## Your Environment

* VGMTrans version used: Qt-branch, commit f88567f (This issue has persisted ever since support for this format was added. I didn't have a GitHub account at the time, and just hoped that someone would eventually notice.)
* Operating System and version: Windows 10 Build 10.0.17134.345
* Game ROM SHA-256 or music dump release name (Zophar, VGMRips, etc): BGM.TVB is 900FA480C28872A2B4F0A27A251AAC928D8BA0FDE104CB2C268BBB2DA17343D2

**Comment by TheShinyCookie** [comment]
Another thing that I've noticed in Road Trip Adventure...

My version of the game, PAL, seems to have more sustain/reverb/whatever than the video that I Iinked here, which some comments say that it's a regional thing. Maybe the decay rate of instruments is affected by the vertical refresh rate, and they didn't bother to change it?

**Comment by sykhro** [comment]
That would be beyond weird. I'm currently investigating this, it seems like we are missing some commands

**Comment by Lunallade** [comment]
Has this been fixed? i love the soundfont and i plan to make music with it.

**Comment by TheShinyCookie** [comment]
Testing this again in the refactor branch, the broken version of the music is now played in the program itself, without having to export.

**Comment by Lunallade** [comment]
meaning?

**Comment by TheShinyCookie** [comment]
Instead of the corruption only occuring when exporting into a MIDI and SF2, now it happens when previewing as well.

**Comment by Lunallade** [comment]
Bugger

**Comment by Meerkov** [comment]
> sustain
I don't think that SF2 or DLS sound fonts support the "Sustain" function of ADSR. (See PSX SPU.h for comments to this effect that the sustain field is currently ignored).

**Comment by ski-freak** [comment]
Did you get the road trip adventure sound font working? if so can you upload it?

**Comment by aEoNNzR** [comment]
When trying to convert a .TVB file into a .SF2 or .DLS, nothing seems to happen on my end. Also looking for the road trip adventure sound font
